partcad/src/partcad/project_factory_git.py

```
@@ -86,82 +113,113 @@ def _clone_or_update_repo(self, repo_url, cache_dir=None):
+                        # Check if the error message matches any of the patterns
+                        if any(re.search(pattern, str(e)) for pattern in git_error_patterns) and attempt < max_retries:
+                            pc_logging.warning(
+                                "Failed to clone repo. Retrying (%d/%d) in %d secs...",
+                                attempt + 1,
+                                max_retries,
+                                patience,
+                            )
+                            time.sleep(patience)
+                        else:
+                            pc_logging.error("Failed to clone repo after %d retries", attempt)
+                            raise RuntimeError("Failed to clone repo: %s" % e)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)